pyomo/core/expr/numvalue.py

```diff
@@ -180,18 +180,12 @@ def value(obj, exception=True):
     """
     if obj.__class__ in native_types:
         return obj
+    if obj.__class__ is NumericConstant:
+        if obj.value is None:
+            raise ValueError("No value for uninitialized NumericConstant object %s" % (obj.name,))
```

Comment:
Bug: this raises an exception regardless of the value of the "exception" flag.

Editorial: separately, why do we have the exception flag?